docs/guides/08-interactivity-events.md

```diff
@@ -0,0 +1,411 @@
+
+Regarding to the `updated` event, it can be useful for some other cases, where the layer's viz changes, for example when you are building an animation. See the [Playing with animations guide](/developers/carto-vl/guides/playing-with-animations).
+
+If you check your work now, it should look like this:
+<div class="example-map">
+    <iframe
+        id="guides-interactivity-step-2"
+        src="/developers/carto-vl/examples/maps/guides/interactivity/step-2.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+
+### Using dynamic variables
+*Variables* are a way to store and reuse expressions, and that can definitively help you when adding interactions to your visualization, so let's practice a bit with them.
+
+#### Variables without properties
+First you are going to add a variable whose value depends solely on the current map extent. Replace your current `const viz = new carto.Viz();`, with this code that grabs the current displayed features using the *String API*:
+```js
+const viz = new carto.Viz(`
+    @currentFeatures: viewportFeatures()
+`);
+```
+
+And then you should add this to handle the updates after you change the map's extent:
+```js
+const displayNumberOfCities = () => {
+    const numberOfFeatures = viz.variables.currentFeatures.value.length;
+    console.log(`Now you can see ${numberOfFeatures} cities`);
+};
+layer.on('updated', displayNumberOfCities);
+```
+> Notice how the variable can be accessed directly from the `carto.Viz` object, inside its `variables` array, without the `@` symbol. Its content is accessible using `.value`, and this is possible cause the expression has no `properties` related to the features themselves.
+
+You can imagine `layer:updated` event as a kind of `layer:viz-updated` event, notifying you whenever something relevant has changed in the viz attached to the layer.
+
+If you want to reduce the number of current `console.log` entries, to better see the new one, you can remove the previous handler on `map:move` with:
+```js
+map.off('move', displayCenter);
+```
+
+You have already advanced a lot in this guide. Now take a small rest and check your work with this:
+<div class="example-map">
+    <iframe
+        id="guides-interactivity-step-3"
+        src="/developers/carto-vl/examples/maps/guides/interactivity/step-3.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### Data-driven variables
+If the data you are interested in for your interaction is a feature `property`, such as the _name_ of the city or its _population_, then you can also use some variables to store them. Those are called _data-driven variables_, cause their values change as you interact with each of the features (in our example, with each city).
+
+To test them you should edit again your viz, so it looks like this:
```

Comment:
To test them you should edit again your viz, so it looks like this --> To test them you should edit again your viz as follows: